### DISCUSSION TITLE
Explanation of Ember driver

### DISCUSSION THREAD

**Discussion by n1mda** [Koenkk/zigbee2mqtt#23218] [General]
Hello,

Starting zigbee2mqtt outputs a warning in the console that the ezsp driver will be deprecated for ember:
```
warning: 	zh:ezsp: Deprecated driver 'ezsp' currently in use, 'ember' will become the officially supported EmberZNet driver in next release. If using Zigbee2MQTT see https://github.com/Koenkk/zigbee2mqtt/discussions/21462
```

I switched to ember, and most things work, apart from devices from Develco, there are already issues about this (e.g. [#22492](https://github.com/Koenkk/zigbee2mqtt/issues/22492))

A [comment](https://github.com/Koenkk/zigbee2mqtt/issues/22492#issuecomment-2143987929) from @Koenkk states that this is a device issue.
However, Develco themself state that the devices are [compatible](https://frient.com/compatible-hubs/) with plenty of Zigbee stacks (including Z2M), and it works as expected using the ezsp driver. So it feels a bit misleading to say it's a device issue and not an ember issue if the device works with _everything_ else _but_ ember.

I have tried searching for documentation about ember and why it is replacing ezsp but have not found anything official.
What will be the official deprecation of ezsp, when will it take place? Will new devices not be supported in ezsp? Trying to understand if it's dumb to remain on ezsp (given it will be replaced in the future).
I'm also wondering if the "device" issue with Develco will be "fixed/worked-around" in ember, or if zigbee2mqtt will drop support for Develco entirely?

Thank you very much for a great piece of software!

**Comment by Koenkk** [comment]
> What will be the official deprecation of ezsp, when will it take place? 

Currently there is no date for the deprecation, although the implementation of it will not be updated anymore

> Will new devices not be supported in ezsp?

New devices will definitely be supported, as device drivers are not tied to the adapter drivers

> I'm also wondering if the "device" issue with Develco will be "fixed/worked-around" in ember, or if zigbee2mqtt will drop support for Develco entirely?

It's unclear yet what exactly causes this, maybe it will improve with the new V8 firmware release.

**Reply by n1mda** [reply]
Thank you @Koenkk 

> Currently there is no date for the deprecation, although the implementation of it will not be updated anymore

Is there any risk that future versions of z2m will not work with ezsp at all?

> It's unclear yet what exactly causes this, maybe it will improve with the new V8 firmware release.

Do you have any references to the V8 release of the firmware? I don't see any discussions about it in [the repo](https://github.com/darkxst/silabs-firmware-builder)

**Reply by Koenkk** [reply]
> Is there any risk that future versions of z2m will not work with ezsp at all?

Not yet, it will be kept to keep support for V6

> Do you have any references to the V8 release of the firmware? I don't see any discussions about it in [the repo](https://github.com/darkxst/silabs-firmware-builder)

See https://github.com/Koenkk/zigbee2mqtt/discussions/23224
